x-pack/legacy/plugins/triggers_actions_ui/np_ready/public/application/components/builtin_action_types/slack.tsx

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import React, { Fragment } from 'react';
+import {
+  EuiFieldText,
+  EuiTextArea,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiButtonIcon,
+  EuiFormRow,
+  EuiLink,
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  ActionTypeModel,
+  ActionConnectorFieldsProps,
+  ActionConnector,
+  ValidationResult,
+  ActionParamsProps,
+} from '../../../types';
+
+export function getActionType(): ActionTypeModel {
+  return {
+    id: '.slack',
+    iconClass: 'logoSlack',
+    selectMessage: i18n.translate(
+      'xpack.triggersActionsUI.components.builtinActionTypes.slackAction.selectMessageText',
+      {
+        defaultMessage: 'Send a message to a Slack channel or user.',
+      }
+    ),
+    validateConnector: (action: ActionConnector): ValidationResult => {
+      const validationResult = { errors: {} };
+      const errors = {
+        webhookUrl: new Array<string>(),
+      };
+      validationResult.errors = errors;
+      if (!action.secrets.webhookUrl) {
+        errors.webhookUrl.push(
+          i18n.translate(
+            'xpack.triggersActionsUI.components.builtinActionTypes.slackAction.error.requiredWebhookUrlText',
+            {
+              defaultMessage: 'Webhook URL is required.',
+            }
+          )
+        );
+      }
+      return validationResult;
+    },
+    validateParams: (actionParams: any): ValidationResult => {
+      const validationResult = { errors: {} };
```

Comment:
`message` should be checked to ensure it's not null/undefined/a least one character